src/main/java/codechicken/nei/BookmarkPanel.java

```diff
@@ -91,6 +230,19 @@ public int indexOf(ItemStack stackA, BookmarkRecipeId recipeId)
     
             return -1;
         }
+            
+        protected int getSlotIndex(int idx)
+        {
+            List<Integer> mask = getMask();
+
+            for (int i = 0; i < mask.size(); i++) {
+                if (mask.get(i) != null && mask.get(i) == idx) {
```

Comment:
```suggestion
                if (mask.get(i) != null && mask.get(i).intValue() == idx) {
```
Boxed integers might be different references even if they have the same value